ctru-rs/src/services/cam.rs

```diff
@@ -21,191 +20,165 @@ pub struct Cam {
 }
 
-bitflags! {
-    /// A set of flags to be passed to [Camera::set_frame_rate]
-    #[derive(Default)]
-    pub struct CamFrameRate: u32 {
-        const RATE_15       = ctru_sys::FRAME_RATE_15;
-        const RATE_15_TO_5  = ctru_sys::FRAME_RATE_15_TO_5;
-        const RATE_15_TO_2  = ctru_sys::FRAME_RATE_15_TO_2;
-        const RATE_10       = ctru_sys::FRAME_RATE_10;
-        const RATE_8_5      = ctru_sys::FRAME_RATE_8_5;
-        const RATE_5        = ctru_sys::FRAME_RATE_5;
-        const RATE_20       = ctru_sys::FRAME_RATE_20;
-        const RATE_20_TO_5  = ctru_sys::FRAME_RATE_20_TO_5;
-        const RATE_30       = ctru_sys::FRAME_RATE_30;
-        const RATE_30_TO_5  = ctru_sys::FRAME_RATE_30_TO_5;
-        const RATE_15_TO_10 = ctru_sys::FRAME_RATE_15_TO_10;
-        const RATE_20_TO_10 = ctru_sys::FRAME_RATE_20_TO_10;
-        const RATE_30_TO_10 = ctru_sys::FRAME_RATE_30_TO_10;
-    }
+/// Flag to pass to [Camera::set_frame_rate]
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum FrameRate {
+    Fps15 = ctru_sys::FRAME_RATE_15,
+    Fps15To5 = ctru_sys::FRAME_RATE_15_TO_5,
+    Fps15To2 = ctru_sys::FRAME_RATE_15_TO_2,
+    Fps10 = ctru_sys::FRAME_RATE_10,
+    Fps8_5 = ctru_sys::FRAME_RATE_8_5,
+    Fps5 = ctru_sys::FRAME_RATE_5,
+    Fps20 = ctru_sys::FRAME_RATE_20,
+    Fps20To5 = ctru_sys::FRAME_RATE_20_TO_5,
+    Fps30 = ctru_sys::FRAME_RATE_30,
+    Fps30To5 = ctru_sys::FRAME_RATE_30_TO_5,
+    Fps15To10 = ctru_sys::FRAME_RATE_15_TO_10,
+    Fps20To10 = ctru_sys::FRAME_RATE_20_TO_10,
+    Fps30To10 = ctru_sys::FRAME_RATE_30_TO_10,
 }
 
-bitflags! {
-    /// A set of flags to be passed to [Camera::set_white_balance] or
-    /// [Camera::set_white_balance_without_base_up]
-    #[derive(Default)]
-    pub struct CamWhiteBalance: u32 {
-        const AUTO  = ctru_sys::WHITE_BALANCE_AUTO;
-        const BALANCE_3200K = ctru_sys::WHITE_BALANCE_3200K;
-        const BALANCE_4150K = ctru_sys::WHITE_BALANCE_4150K;
-        const BALANCE_5200K = ctru_sys::WHITE_BALANCE_5200K;
-        const BALANCE_6000K = ctru_sys::WHITE_BALANCE_6000K;
-        const BALANCE_7000K = ctru_sys::WHITE_BALANCE_7000K;
-
-        const NORMAL                  = ctru_sys::WHITE_BALANCE_NORMAL;
-        const TUNGSTEN                = ctru_sys::WHITE_BALANCE_TUNGSTEN;
-        const WHITE_FLUORESCENT_LIGHT = ctru_sys::WHITE_BALANCE_WHITE_FLUORESCENT_LIGHT;
-        const DAYLIGHT                = ctru_sys::WHITE_BALANCE_DAYLIGHT;
-        const CLOUDY                  = ctru_sys::WHITE_BALANCE_CLOUDY;
-        const HORIZON                 = ctru_sys::WHITE_BALANCE_HORIZON;
-        const SHADE                   = ctru_sys::WHITE_BALANCE_SHADE;
-    }
+/// Flag to pass to [Camera::set_white_balance] or
+/// [Camera::set_white_balance_without_base_up]
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum WhiteBalance {
+    /// Normal
+    Auto = ctru_sys::WHITE_BALANCE_AUTO,
+    /// Tungsten
+    Type3200K = ctru_sys::WHITE_BALANCE_3200K,
```

Comment:
`Type` prefix here seems redundant to me, but I guess you can't just use numbered values. Maybe `Temp3200K` instead?

Reply:
I searched far and wide for a good prefix, let's go with `Temp`.